irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -111,12 +118,14 @@ OnDemandOrderingServiceImpl::emitProposal() {
   TransactionType current_tx;
   using ProtoTxType = shared_model::proto::Transaction;
   std::vector<TransactionType> collection;
+  std::unordered_set<std::string> inserted;
 
   // outer method should guarantee availability of at least one transaction in
   // queue, also, code shouldn't fetch all transactions from queue. The rest
   // will be lost.
   while (current_proposal_.second.try_pop(current_tx)
-         and collection.size() < transaction_limit_) {
+         and collection.size() < transaction_limit_
+         and inserted.insert(current_tx->hash().hex()).second) {
```

Comment:
Do we clear up inserted set somewhere? It can grow large after a while. Also what about peer restart?

Reply:
It is local to the method, so it is cleared after each proposal.